--- origsrc/libmbfl/filters/Makefile.am	2008-05-07 15:45:39.000000000 -0500
+++ src/libmbfl/filters/Makefile.am	2008-05-07 15:59:13.015625000 -0500
@@ -1,6 +1,6 @@
 EXTRA_DIST=Makefile.bcc32 mk_sb_tbl.awk
 noinst_LTLIBRARIES=libmbfl_filters.la
-INCLUDES=-I../mbfl
+INCLUDES=-I$(top_srcdir)/mbfl
 libmbfl_filters_la_LDFLAGS=-version-info $(SHLIB_VERSION)
 libmbfl_filters_la_SOURCES=mbfilter_cp936.c \
 	mbfilter_hz.c \
@@ -165,65 +165,65 @@
 8859-14.TXT 8859-15.TXT 8859-16.TXT:
 	$(FETCH_VIA_FTP) ftp://ftp.unicode.org/Public/MAPPINGS/ISO8859/$@
 
-unicode_table_iso8859_1.h: mk_sb_tbl.awk
+unicode_table_iso8859_1.h: mk_sb_tbl.awk 8859-1.TXT
 	$(AWK) -v TABLE_NAME=iso8859_1_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_1_H -f mk_sb_tbl.awk 8859-1.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_1_H -f $^ > $@
 
-unicode_table_iso8859_2.h: mk_sb_tbl.awk
+unicode_table_iso8859_2.h: mk_sb_tbl.awk 8859-2.TXT
 	$(AWK) -v TABLE_NAME=iso8859_2_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_2_H -f mk_sb_tbl.awk 8859-2.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_2_H -f $^ > $@
 
-unicode_table_iso8859_3.h: mk_sb_tbl.awk
+unicode_table_iso8859_3.h: mk_sb_tbl.awk 8859-3.TXT
 	$(AWK) -v TABLE_NAME=iso8859_3_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_3_H -f mk_sb_tbl.awk 8859-3.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_3_H -f $^ > $@
 
-unicode_table_iso8859_4.h: mk_sb_tbl.awk 
+unicode_table_iso8859_4.h: mk_sb_tbl.awk 8859-4.TXT
 	$(AWK) -v TABLE_NAME=iso8859_4_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_4_H -f mk_sb_tbl.awk 8859-4.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_4_H -f $^ > $@
 
-unicode_table_iso8859_5.h: mk_sb_tbl.awk 
+unicode_table_iso8859_5.h: mk_sb_tbl.awk 8859-5.TXT
 	$(AWK) -v TABLE_NAME=iso8859_5_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_5_H -f mk_sb_tbl.awk 8859-5.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_5_H -f $^ > $@
 
-unicode_table_iso8859_6.h: mk_sb_tbl.awk 
+unicode_table_iso8859_6.h: mk_sb_tbl.awk 8859-6.TXT
 	$(AWK) -v TABLE_NAME=iso8859_6_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_6_H -f mk_sb_tbl.awk 8859-6.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_6_H -f $^ > $@
 
-unicode_table_iso8859_7.h: mk_sb_tbl.awk 
+unicode_table_iso8859_7.h: mk_sb_tbl.awk 8859-7.TXT
 	$(AWK) -v TABLE_NAME=iso8859_7_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_7_H -f mk_sb_tbl.awk 8859-7.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_7_H -f $^ > $@
 
-unicode_table_iso8859_8.h: mk_sb_tbl.awk 
+unicode_table_iso8859_8.h: mk_sb_tbl.awk 8859-8.TXT
 	$(AWK) -v TABLE_NAME=iso8859_8_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_8_H -f mk_sb_tbl.awk 8859-8.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_8_H -f $^ > $@
 
-unicode_table_iso8859_9.h: mk_sb_tbl.awk 
+unicode_table_iso8859_9.h: mk_sb_tbl.awk 8859-9.TXT
 	$(AWK) -v TABLE_NAME=iso8859_9_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_9_H -f mk_sb_tbl.awk 8859-9.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_9_H -f $^ > $@
 
-unicode_table_iso8859_10.h: mk_sb_tbl.awk 
+unicode_table_iso8859_10.h: mk_sb_tbl.awk 8859-10.TXT
 	$(AWK) -v TABLE_NAME=iso8859_10_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_10_H -f mk_sb_tbl.awk 8859-10.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_10_H -f $^ > $@
 
-unicode_table_iso8859_11.h: mk_sb_tbl.awk 
+unicode_table_iso8859_11.h: mk_sb_tbl.awk 8859-11.TXT
 	$(AWK) -v TABLE_NAME=iso8859_11_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_11_H -f mk_sb_tbl.awk 8859-11.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_11_H -f $^ > $@
 
-unicode_table_iso8859_13.h: mk_sb_tbl.awk 
+unicode_table_iso8859_13.h: mk_sb_tbl.awk 8859-13.TXT
 	$(AWK) -v TABLE_NAME=iso8859_13_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_13_H -f mk_sb_tbl.awk 8859-13.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_13_H -f $^ > $@
 
-unicode_table_iso8859_14.h: mk_sb_tbl.awk 
+unicode_table_iso8859_14.h: mk_sb_tbl.awk 8859-14.TXT
 	$(AWK) -v TABLE_NAME=iso8859_14_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_14_H -f mk_sb_tbl.awk 8859-14.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_14_H -f $^ > $@
 
-unicode_table_iso8859_15.h: mk_sb_tbl.awk 
+unicode_table_iso8859_15.h: mk_sb_tbl.awk 8859-15.TXT
 	$(AWK) -v TABLE_NAME=iso8859_15_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_15_H -f mk_sb_tbl.awk 8859-15.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLE_ISO8859_15_H -f $^ > $@
 
-unicode_table_iso8859_16.h: mk_sb_tbl.awk 
+unicode_table_iso8859_16.h: mk_sb_tbl.awk 8859-16.TXT
 	$(AWK) -v TABLE_NAME=iso8859_16_ucs_table \
-           -v IFNDEF_NAME=UNICODE_TABLEISO8859_16_H -f mk_sb_tbl.awk 8859-16.TXT > $@
+           -v IFNDEF_NAME=UNICODE_TABLEISO8859_16_H -f $^ > $@
 
 unidata: 8859-1.TXT 8859-2.TXT 8859-3.TXT 8859-4.TXT 8859-5.TXT 8859-6.TXT \
 8859-7.TXT 8859-8.TXT 8859-9.TXT 8859-10.TXT 8859-11.TXT 8859-13.TXT \
--- origsrc/libmbfl/mbfl/Makefile.am	2008-05-07 15:45:28.000000000 -0500
+++ src/libmbfl/mbfl/Makefile.am	2008-05-07 16:02:58.218750000 -0500
@@ -1,5 +1,6 @@
 EXTRA_DIST=Makefile.bcc32 mk_eaw_tbl.awk
 lib_LTLIBRARIES=libmbfl.la
+INCLUDES=-I$(top_srcdir)
 libmbfl_la_SOURCES=mbfilter.c \
 	mbfl_string.c \
 	mbfl_language.c \
@@ -16,7 +17,7 @@
 libmbfl_filters_la=../filters/libmbfl_filters.la
 libmbfl_nls_la=../nls/libmbfl_nls.la
 libmbfl_la_LIBADD=$(libmbfl_filters_la) $(libmbfl_nls_la)
-libmbfl_la_LDFLAGS=-version-info $(SHLIB_VERSION)
+libmbfl_la_LDFLAGS=-version-info $(SHLIB_VERSION) -no-undefined
 libmbfl_includedir=$(includedir)/mbfl
 libmbfl_include_HEADERS=mbfilter.h \
 	mbfl_consts.h \
@@ -35,8 +36,8 @@
 
 mbfilter.c: eaw_table.h
 
-eaw_table.h: mk_eaw_tbl.awk
-	$(AWK) -v TABLE_NAME=mbfl_eaw_table -f mk_eaw_tbl.awk EastAsianWidth.txt > $@
+eaw_table.h: mk_eaw_tbl.awk EastAsianWidth.txt
+	$(AWK) -v TABLE_NAME=mbfl_eaw_table -f $^ > $@
 
 EastAsianWidth.txt:
 	$(FETCH_VIA_FTP) ftp://ftp.unicode.org/Public/UNIDATA/EastAsianWidth.txt
--- origsrc/libmbfl/nls/Makefile.am	2008-05-07 15:45:29.000000000 -0500
+++ src/libmbfl/nls/Makefile.am	2008-05-07 15:51:27.359375000 -0500
@@ -1,6 +1,6 @@
 EXTRA_DIST=Makefile.bcc32
 noinst_LTLIBRARIES=libmbfl_nls.la
-INCLUDES=-I../mbfl
+INCLUDES=-I$(top_srcdir)/mbfl
 libmbfl_nls_la_LDFLAGS=-version-info $(SHLIB_VERSION)
 libmbfl_nls_la_SOURCES=nls_ja.c \
 	nls_de.c \
--- origsrc/libmbfl/tests/Makefile.am	2008-05-07 15:45:31.000000000 -0500
+++ src/libmbfl/tests/Makefile.am	2008-05-07 16:04:09.406250000 -0500
@@ -1,5 +1,6 @@
 SUBDIRS=conv_encoding.tests conv_kana.tests strwidth.tests strcut.tests
 noinst_PROGRAMS=conv_encoding conv_kana strwidth strcut
+INCLUDES=-I$(top_srcdir)
 conv_encoding_SOURCES=conv_encoding.c
 conv_encoding_LDADD=../mbfl/libmbfl.la
 conv_kana_SOURCES=conv_kana.c
